### PR TITLE
[T4A2][T16-B2]ZhaoDingfan

### PR DESCRIPTION
**Pull request by dfz2019**

### CODE REVIEW

**Comment by dfz2019** [comment]
Dear tutor, I have a doubt that how can I remove the first two commits there? It is just one commit by mistake and the revert commit.

**Comment by dfz2019** [comment]
Ready for review.

**Comment by PierceAndy** [comment]
@ZhaoDingfan perhaps you can try googling for a solution. I'm sure that it's a well encountered issue. I believe it was also touched upon in lecture 4.

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ public Command parseCommand(String userInput) {
      * @param args full command args string
      * @return the prepared command
      */
-    private Command prepareAdd(String args){
+    private static Command prepareAdd(String args){
```
Comment (PierceAndy) [inline src/seedu/addressbook/parser/Parser.java:99]: Good job on making these private methods static as well. As `parseCommand` was made static, any methods that it depends on has to be made static as well.

File: src/seedu/addressbook/data/person/Person.java [hunk 2]
```diff
@@ -25,6 +27,8 @@ public Person(Name name, Phone phone, Email email, Address address, UniqueTagLis
         this.email = email;
         this.address = address;
         this.tags = new UniqueTagList(tags); // protect internal tags from changes in the arg list
+        nextSequenceNumber++;
+        sequenceNumber = nextSequenceNumber;
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/person/Person.java:31]: Semantically, the following sequence would make more sense:

```
private static int nextSequenceNumber = 1;
sequenceNumber = nextSequenceNumber;
nextSequenceNumber++;
```

By the way,

```
sequenceNumber = nextSequenceNumber;
nextSequenceNumber++;
```

can be shortened into
`sequenceNumber = nextSequenceNumber++;`

And likewise

```
nextSequenceNumber++;
sequenceNumber = nextSequenceNumber;
```

can be shortened into
`sequenceNumber = ++nextSequenceNumber;`

**Comment by PierceAndy** [comment]
Remember to update the copy constructor as well. When modifying a constructor, remember to check if other constructors need to be changed too.

@ZhaoDingfan Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**